base/HHVMDaemon.php

```diff
@@ -151,15 +150,18 @@ public function start(): void {
 
       invariant(is_dir($sourceRoot), 'Could not find valid source root');
 
-      $files = Vector{};
       $dir_iter = new RecursiveDirectoryIterator($sourceRoot);
       $iter     = new RecursiveIteratorIterator($dir_iter);
       foreach ($iter as $path => $_) {
-        if (is_file($path)) {
-          $files->add(ltrim(substr($path, strlen($sourceRoot)), '/'));
+        // Source files not ending in .php need to be specifically included
+        if (is_file($path) && substr($path, -4) !== '.php') {
+          $contents = file_get_contents($path);
```

Comment:
file_get_contents supports a maxlen argument.  You only care about the first 4 bytes, so no point in reading the whole thing.